src/lib/components/data-graphic/marks/Line.svelte

```diff
@@ -67,13 +69,65 @@
       pathDefined,
     });
   }
+
+  $: totalTravelDistance = data
+    .map((di, i) => {
+      if (i === data.length - 1) {
+        return 0;
+      }
+      const max = Math.max(
+        $yScale(data[i + 1][yAccessor]),
+        $yScale(data[i][yAccessor])
+      );
+      const min = Math.min(
+        $yScale(data[i + 1][yAccessor]),
+        $yScale(data[i][yAccessor])
+      );
+      return Math.abs(max - min);
+    })
+    .reduce((acc, v) => acc + v, 0);
+
+  let lineDensity = 0.05;
+  let devicePixelRatio = 3;
+
+  $: lineDensity = Math.min(
+    1,
+    /** to determine the stroke width of the path, let's look at
+     * the bigger of two values:
+     * 1. the "y-ish" distance travelled
+     * the inverse of "total travel distance", which is the Y
+     * gap size b/t successive points divided by the zoom window size;
+     * 2. time series length / available X pixels
+     * the time series divided by the total number of pixels in the existing
+     * zoom window.
+     *
+     * These heuristics could be refined, but this seems to provide a reasonable approximation for
+     * the stroke width. (1) excels when lots of successive points are close together in the Y direction,
+     * whereas (2) excels when a line is very, very noisy (and thus the X direction is the main constraint).
+     */
+    Math.max(
+      2 /
+        (totalTravelDistance /
+          (($xScale.range()[1] - $xScale.range()[0]) *
+            ($config.devicePixelRatio || 3))),
+      (($xScale.range()[1] - $xScale.range()[0]) *
+        ($config.devicePixelRatio || 3) *
+        0.7) /
+        data.length /
+        1.5
+    )
+  );
+
+  onMount(() => {
+    devicePixelRatio = window.devicePixelRatio;
+  });
 </script>
 
 {#if lineFcn}
   <path
     d={lineFcn(yAccessor)(data)}
     stroke={color}
-    stroke-width={lineThickness}
+    stroke-width={lineDensity}
```

Comment:
You like `lineDensity` over `lineThickness`? `lineThickness` seems a bit more intuitive to me

Reply:
I should specify that these are two different values, I copied and pasted something here